src/main/java/org/assertj/guava/internal/RangeSets.java

```diff
@@ -0,0 +1,854 @@
+import org.assertj.core.internal.Objects;
+import org.assertj.core.util.VisibleForTesting;
+
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeSet;
+
+/**
+ * Reusable assertions for <code>{@link RangeSet}</code>s.
+ *
+ * @author Ilya_Koshaleu
+ */
+public class RangeSets {
+
+  private final static RangeSets INSTANCE = new RangeSets();
+
+  @VisibleForTesting
+  Failures failures = Failures.instance();
+
+  /**
+   * Returns singleton instance of this class.
+   * 
+   * @return singleton instance of this class.
+   */
+  public static RangeSets instance() {
+    return INSTANCE;
+  }
+
+  /**
+   * Asserts that the actual {@code RangeSet} has specific {@code size}. 
+   * 
+   * @param info contains information about the assertion.
+   * @param actual the given {@code RangeSet}.
+   * @param size expected size
+   */
+  public void assertHasSize(AssertionInfo info, RangeSet actual, int size) {
+    assertNotNull(info, actual);
+    checkSizes(actual, sizeOf(actual.asRanges()), size, info);
+  }
+
+  /**
+   * Asserts that the given {@code RangeSet} contains the given values.
+   * 
+   * @param info contains information about the assertion.
+   * @param actual the given {@code RangeSet}.
+   * @param values the values that are expected to be in the given {@code RangeSet}.
+   * @throws AssertionError if the actual {@code RangeSet} is {@code null}.
+   * @throws AssertionError if the actual {@code RangeSet} does not contain the given {@code elements}.
+   * @throws IllegalArgumentException if elements are null or elements are empty while actual is not empty.
+   */
+  public void assertContains(AssertionInfo info, RangeSet actual, Comparable[] values) {
+    assertNotNull(info, actual);
+    failIfNull(values);
+    // Should pass if both actual and expected are empty
+    if (actual.isEmpty() && values.length == 0) return;
+    failIfEmpty(values);
+    assertRangeSetContainsGivenValues(info, actual, values);
+  }
+
+  /**
+   * Asserts that the given {@code RangeSet} contains all the given values.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code RangeSet}.
+   * @param values the values that are expected to be in the given {@code RangeSet}.
+   * @throws AssertionError if the actual {@code RangeSet} is {@code null}.
+   * @throws AssertionError if the actual {@code RangeSet} does not contain all the given {@code elements}.
+   * @throws IllegalArgumentException if elements are null or elements are empty while actual is not empty.
+   */
+  public void assertContainsAll(AssertionInfo info, RangeSet actual, Iterable<? extends Comparable> values) {
+    assertNotNull(info, actual);
+    failIfNull(values);
+    // Should pass if both actual and expected are empty
+    if (actual.isEmpty() && !values.iterator().hasNext()) return;
+    failIfEmpty(values);
+    assertRangeSetContainsGivenValues(info, actual, toArray(values));
```

Comment:
convert to a `Comparable[]` so that `assertRangeSetContainsGivenValues` can take `Comparable[]` instead of `Object[]`

Reply:
this comment applies to other method that do the same

Reply:
done

Reply:
done